integration/src/main/java/scripts/api/TdrDatasetsApi.java

```diff
@@ -0,0 +1,166 @@
+  private DatasetModel testDataset;
+
+  private final bio.terra.datarepo.api.DatasetsApi datasetsApi;
+
+  private TdrDatasetsApi(DatarepoClient apiClient) {
+    datasetsApi = new bio.terra.datarepo.api.DatasetsApi(apiClient);
+    // Register a shutdown hook to clean up the test dataset on JVM shutdown. It's done this way
+    // because we don't know which test that uses a dataset will run last, and we don't want to
+    // make and delete a dataset for every test.
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () -> {
+                  if (testDataset != null) {
+                    try {
+                      deleteDataset(testDataset.getId());
+                      testDataset = null;
+                    } catch (Exception e) {
+                      log.error("error deleting dataset during shutdown", e);
+                    }
+                  }
+                }));
+  }
+
+  public static synchronized TdrDatasetsApi createApi(DatarepoClient apiClient) {
+    if (theApi == null) {
+      theApi = new TdrDatasetsApi(apiClient);
+    }
+    return theApi;
+  }
+
+  private DatarepoClient getApiClient() {
+    return (DatarepoClient) datasetsApi.getApiClient();
+  }
+
+  private UUID getBillingProfileId() throws ApiException {
+    ProfilesApi profilesApi = new ProfilesApi(getApiClient());
+    return profilesApi.enumerateProfiles(0, 100).getItems().stream()
+        .filter(model -> model.getProfileName().startsWith(TEST_BILLING_PROFILE_PREFIX))
+        .findFirst()
+        .orElseThrow(() -> new RuntimeException("Test billing profile not found"))
+        .getId();
+  }
+
+  private static class Column extends ColumnModel {
+    Column(String name, TableDataType dataType) {
+      setName(name);
+      setDatatype(dataType);
+    }
+  }
+
+  private static class Table extends TableModel {
+    Table(String name, ColumnModel... columns) {
+      setName(name);
+      setColumns(Arrays.asList(columns));
+    }
+  }
+
+  private DatasetRequestModel createDatasetRequestModel(UUID billingProfileId) {
+    var schema =
+        new DatasetSpecificationModel()
+            .tables(
+                List.of(
+                    new Table(
+                        "participant",
+                        new Column("participant_id", TableDataType.STRING),
+                        new Column("biological_sex", TableDataType.STRING),
+                        new Column("age", TableDataType.INTEGER)),
+                    new Table(
+                        "sample",
+                        new Column("sample_id", TableDataType.STRING),
+                        new Column("participant_id", TableDataType.STRING),
+                        new Column("files", TableDataType.STRING).arrayOf(true),
+                        new Column("type", TableDataType.STRING))));
+    return new DatasetRequestModel()
+        .name("catalog_integration_test_" + System.currentTimeMillis())
```

Comment:
if we're trying to guard against locking problems related to multiple tests running at the same time, is there a way to incorporate a uuid in here instead of the system time?

Reply:
I tried using uuid at first but hyphens are disallowed so I switched to millis. Do you really think we'll have a collision of milliseconds in practice? That seems unlikely, but I could remove hyphens from a UUID instead of using milliseconds.